-1

Tested at: 24556e319541e3d00c4fbe39decb4a6b2dfd6240


The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
https://github.com/cms-sw/cmssw/commit/af8dcd515392d4e724f5f36a5b52d96817619d88
https://github.com/cms-sw/cmssw/commit/1676bb9288c29da380b2fb0aaa8ffa0880392ba5
https://github.com/cms-sw/cmssw/commit/335ffe902a08203a891d161ca23887354fd26a63
https://github.com/cms-sw/cmssw/commit/62fb52534a3c5bf8082c714eea35bf671356679d
https://github.com/cms-sw/cmssw/commit/5b32db8d0ab8133d934855a2b0a5a66f5d67574d
https://github.com/cms-sw/cmssw/commit/de1028ca180aecdd816ccab626f9ecd654713f9e
https://github.com/cms-sw/cmssw/commit/a69fe89cecdb98ceb434f115b92edaeebd1e596d
https://github.com/cms-sw/cmssw/commit/db641316d5cbc42ac92ee74adae6dd5bf095f65d
You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-16891/16826/git-log-recent-commits
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-16891/16826/git-merge-result

You can see the results of the tests here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-16891/16826/summary.html

I found follow errors while testing this PR

**Failed tests:  Build**

* **Build**:

I found an error when building: 
 
 <pre>>> Compiling  /build/cmsbuild/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_0_X_2016-12-06-1100/src/RecoTracker/TkMSParametrization/src/MSLayer.cc 
>> Compiling  /build/cmsbuild/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_0_X_2016-12-06-1100/src/RecoTracker/TkMSParametrization/src/MSLayersKeeperX0DetLayer.cc 
>> Compiling  /build/cmsbuild/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_0_X_2016-12-06-1100/src/RecoTracker/TkMSParametrization/src/MSLayersAtAngle.cc 
>> Compiling  /build/cmsbuild/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_0_X_2016-12-06-1100/src/RecoTracker/TkMSParametrization/src/MultipleScatteringParametrisation.cc 
In file included from input_line_9:45:
/build/cmsbuild/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_0_X_2016-12-06-1100/src/RecoTracker/TkTrackingRegions/interface/TrackingRegion.h:46:33: error: no type named 'ConstRecHitPointer' in 'SeedingLayerSetsHits'
  typedef SeedingLayerSetsHits::ConstRecHitPointer Hit;
          ~~~~~~~~~~~~~~~~~~~~~~^
/build/cmsbuild/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_0_X_2016-12-06-1100/src/RecoTracker/TkTrackingRegions/interface/TrackingRegion.h:47:33: error: no type named 'Hits' in 'SeedingLayerSetsHits'
  typedef SeedingLayerSetsHits::Hits Hits;
          ~~~~~~~~~~~~~~~~~~~~~~^
</pre>


The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
https://github.com/cms-sw/cmssw/commit/af8dcd515392d4e724f5f36a5b52d96817619d88
https://github.com/cms-sw/cmssw/commit/1676bb9288c29da380b2fb0aaa8ffa0880392ba5
https://github.com/cms-sw/cmssw/commit/335ffe902a08203a891d161ca23887354fd26a63
https://github.com/cms-sw/cmssw/commit/62fb52534a3c5bf8082c714eea35bf671356679d
https://github.com/cms-sw/cmssw/commit/5b32db8d0ab8133d934855a2b0a5a66f5d67574d
https://github.com/cms-sw/cmssw/commit/de1028ca180aecdd816ccab626f9ecd654713f9e
https://github.com/cms-sw/cmssw/commit/a69fe89cecdb98ceb434f115b92edaeebd1e596d
https://github.com/cms-sw/cmssw/commit/db641316d5cbc42ac92ee74adae6dd5bf095f65d
You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-16891/16826/git-log-recent-commits
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-16891/16826/git-merge-result

--- a/TrackingTools/TransientTrackingRecHit/interface/SeedingLayerSetsHits.h
+++ b/TrackingTools/TransientTrackingRecHit/interface/SeedingLayerSetsHits.h
@@ -11,7 +11,6 @@
 
 class DetLayer;
 
-#if !defined(__CINT__) && !defined(__MAKECINT__) && !defined(__REFLEX__)
 /**
  * Class to store TransientTrackingRecHits, names, and DetLayer
  * pointers of each ctfseeding::SeedingLayer as they come from
@@ -249,17 +248,5 @@ private:
    */
   OwnedHits rechits_;
 };
-
-
-#else
-class SeedingLayerSetsHits {
-private:
-  SeedingLayerSetsHits(SeedingLayerSetsHits const&){} 
-  SeedingLayerSetsHits& operator=(SeedingLayerSetsHits const&){return *this;}
-
-  std::vector<BaseTrackerRecHit const*> rechits_;
-
-};
-#endif
 
 #endif

